### ISSUE TITLE
recast and babel printing two different ASTs

### ISSUE THREAD

**Issue by cangoektas** [benjamn/recast#480]
First of all, this library is pure magic! I love it! 😍 

I discovered an interesting behaviour when using `babylon` with custom options and transforming an AST to use the experimental `export-default-from` syntax. I'm trying to transform this:

```js
export { default as foo } from 'foo';
```

into this:

```js
export foo from 'foo';
```

When I use babel to print the AST, I get what I expect. However, when using `recast.print` the result is:

```js
export { foo } from 'foo';
```

You can use the following script to reproduce the issue:

```js
const babylon = require("babylon");
const recast = require("recast");
const j = require("jscodeshift");
const generate = require("@babel/generator").default;

const recastOptions = {
  parser: {
    parse(source) {
      return babylon.parse(source, {
        sourceType: "module",
        plugins: ["exportDefaultFrom"]
      });
    }
  }
};
const code = "export { default as foo } from 'foo';";
const ast = recast.parse(code, recastOptions);

// Replace ExportSpecifier with ExportDefaultSpecifier
j(ast)
  .find(j.ExportSpecifier)
  .forEach(path => {
    const exportedName = path.node.exported.name;

    return j(path).replaceWith(
      j.exportDefaultSpecifier(j.identifier(exportedName))
    );
  });

const recastOutput = recast.print(ast).code;
const babelOutput = generate(ast).code;

console.log(recastOutput); // export { foo } from 'foo';
console.log(babelOutput); // export foo from 'foo';

```

Is there something I'm missing? Is there a way to transform the AST, so that recast prints the changes I want? Or is there something wrong with the `print` function?

Here's the `package.json` file I used:

```json
{
  "dependencies": {
    "@babel/generator": "^7.0.0-beta.42",
    "babylon": "^7.0.0-beta.42",
    "jscodeshift": "^0.5.0",
    "recast": "^0.14.5"
  }
}
```

**Comment by benjamn** [comment]
Hmm, this looks like a bug in the reprinter. Specifically, `default as foo` seems to be getting replaced with just `foo`, without removing the `{...}`s. The `foo` identifier is definitely what we want, but it needs to replace more than just `default as foo` in the original code. Bugs like this can usually be solved by triggering reprinting of the entire `export` declaration.

Is there any chance you could turn this into a failing test case that I could merge into this repo? If direct AST manipulation is too much work, feel free to add `jscodeshift` as a devDependency.

**Comment by cangoektas** [comment]
Thanks for the quick response! Just created a PR with the failing test 👍

**Comment by cangoektas** [comment]
This issue was fixed in #483
